src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.CSharp.cs

```diff
@@ -742,6 +742,77 @@ public class C
 
                 await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.CSharp, expected: expected, signaturesOnly: signaturesOnly, languageVersion: "Preview", metadataLanguageVersion: "Preview");
             }
+
+            [Theory, CombinatorialData, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+            public async Task TestRequiredProperty(bool signaturesOnly)
+            {
+                var metadataSource = """
+                    public class C { public required int Property { get; set; } }
+                    namespace System.Runtime.CompilerServices
+                    {
+                        public sealed class RequiredMemberAttribute : Attribute { }
+                        public sealed class CompilerFeatureRequiredAttribute : Attribute
+                        {
+                            public CompilerFeatureRequiredAttribute(string featureName)
+                            {
+                                FeatureName = featureName;
+                            }
+                            public string FeatureName { get; }
+                            public bool IsOptional { get; set; }
+                        }
+                    }
```

Comment:
Don't we have a file somewhere where we put constants for stuff like this? I thinki we might have for span/index?

Reply:
Not that I can find.